.github/workflows/release.yml

```diff
@@ -34,115 +44,30 @@ on:
-      - name: set RELEASE_KIND = patch
-        if: ${{ (contains(github.event.pull_request.labels.*.name, 'patch_release')) }}
-        run: |
-          echo "RELEASE_KIND=patch" >> $GITHUB_ENV
-      - name: set RELEASE_REF
-        run: |
-          if [[ "${{ github.event_name }}" = "workflow_dispatch" ]]; then
-            echo "RELEASE_REF=${{ github.ref_name }}" >> $GITHUB_ENV
-          else
-            echo "RELEASE_REF=${{ github.base_ref }}" >> $GITHUB_ENV
-          fi
-      - name: "LATEST_VERSION"
-        run: |
-          if [[ "${{ env.RELEASE_REF }}" = "main" ]]; then
-            echo "LATEST_VERSION=$(git tag | grep "^v" | sed 's/^v//' | sort --version-sort | tail -n 1)" >> $GITHUB_ENV
-          else
-            echo "LATEST_VERSION=$(git tag | grep "^v${{ env.RELEASE_REF }}" | sed 's/^v//' | sort --version-sort | tail -n 1)" >> $GITHUB_ENV
-          fi
-      - name: "default LATEST_VERSION"
-        run: |
-          # default to 0.1.0 when there is no previous tag and on main branch
-          if ([[ -z "${{ env.LATEST_VERSION }}" ]] &&  [[ "${{ env.RELEASE_REF }}" = "main" ]]); then
-            echo "LATEST_VERSION=0.1.0" >> $GITHUB_ENV
-          fi
-      # safeguard
-      - name: RELEASE_REF != NULL
-        run: ([ -n "${{ env.RELEASE_REF }}" ])
-      - name: LATEST_VERSION != NULL
-        run: ([ -n "${{ env.LATEST_VERSION }}" ])
-      - name: RELEASE_KIND != NULL
-        run: ([ -n "${{ env.RELEASE_KIND }}" ])
-      - name: "NEW_VERSION"
-        run: |
-          echo "NEW_VERSION=$(sh .github/enhance_version.sh ${{ env.LATEST_VERSION }} ${{ env.RELEASE_KIND }})" >> $GITHUB_ENV
-      - name: NEW_VERSION != NULL
-        run: ([ -n "${{ env.NEW_VERSION }}" ])
-      - name: set output
-        id: version
-        run: |
-          echo "latest_version=$LATEST_VERSION" >> "$GITHUB_OUTPUT"
-          echo "new_version=$NEW_VERSION" >> "$GITHUB_OUTPUT"
-          echo "release_kind=$RELEASE_KIND" >> "$GITHUB_OUTPUT"
-          echo "release_ref=$RELEASE_REF" >> "$GITHUB_OUTPUT"
-          echo "project=$(echo "${{ github.repository }}" | sed 's/.*\///' )" >> "$GITHUB_OUTPUT"
-
-  build-binaries:
-    needs: calculate_version
-    uses: ./.github/workflows/build-rust.yml
-
-
   release:
     name: release
-    needs: 
-      - build-binaries
-      - calculate_version
     runs-on: "ubuntu-latest"
     env:
-      RELEASE_KIND: ${{needs.calculate_version.outputs.release_kind}}
-      RELEASE_REF: ${{needs.calculate_version.outputs.release_ref}}
-      LATEST_VERSION: ${{needs.calculate_version.outputs.latest_version}}
-      NEW_VERSION: ${{needs.calculate_version.outputs.new_version}}
-      PROJECT: ${{needs.calculate_version.outputs.project}}
+      RELEASE_KIND: ${{inputs.release_kind}}
+      RELEASE_REF: ${{inputs.release_ref}}
+      LATEST_VERSION: ${{inputs.latest_version}}
+      NEW_VERSION: ${{inputs.new_version}}
+      PROJECT: ${{inputs.project}}
     steps:
```

Comment:
I really think you should use the https://github.com/greenbone/actions/tree/main/release action here

Reply:
Last time I checked it didn't match all our needs, hence we have that self written approach. We may change it, however based on the massive restructure I would prefer to do that later.

Reply:
For an example it doesn't seem to create a branch on a major version change. It doesn't support Rust. 

The capability of creating a branch on major version is crucial for us and part of the existing release process. Given the complexity of github action and the inability to validate it on a local machine I really, really think this is not the right PR for requesting release process changes.

Reply:
If you really really want to maintain your specific workflows on your own you can of course do whatever you want.

Reply:
we already do that, that's not changed within this PR: https://github.com/greenbone/openvas-scanner/blob/main/.github/workflows/release.yml

I would love to switch, but again, we can't as that action doesn't cover everything we need. I didn't say you're wrong I did say this is not the time to bring it up.

Reply:
I could see you doing - instead of maintaining your own:
a) add the desired features to the action (contribution)
b) ask DevOps to do a)

Would that solve your problem?

Reply:
yes, b. would be, but again: not the right PR to request that. Can we please go down the official route instead of discussing on a PR that DOES NOT change the release process.